API/ev3_constants.h

```diff
@@ -201,31 +208,55 @@ extern "C" {
  * \sa ButtonPressed(), ButtonState(), ReadButtonEx(),
  * @{
  */
+
 // Button IDs
-#define BUTTON_ID_UP     0x01 //!< The up button.
-#define BUTTON_ID_ENTER  0x02 //!< The enter button.
-#define BUTTON_ID_DOWN   0x04 //!< The down button.
-#define BUTTON_ID_RIGHT  0x08 //!< The right button.
-#define BUTTON_ID_LEFT   0x10 //!< The left button.
-#define BUTTON_ID_ESCAPE 0x20 //!< The exit (escape) button.
-#define BUTTON_ID_ALL    0x3f //!< All buttons mask.
-
-#define BTN1 BUTTON_ID_ESCAPE //!< The exit (escape) button.
-#define BTN2 BUTTON_ID_RIGHT  //!< The right button.
-#define BTN3 BUTTON_ID_LEFT   //!< The left button.
-#define BTN4 BUTTON_ID_ENTER  //!< The enter button.
-#define BTN5 BUTTON_ID_UP     //!< The up button.
-#define BTN6 BUTTON_ID_DOWN   //!< The down button.
-
-#define BTNEXIT   BTN1 //!< The exit (escape) button.
-#define BTNRIGHT  BTN2 //!< The right button.
-#define BTNLEFT   BTN3 //!< The left button.
-#define BTNCENTER BTN4 //!< The enter button.
-#define BTNUP     BTN5 //!< The up button.
-#define BTNDOWN   BTN6 //!< The down button.
+
+//! Button identifier for normal uses.
+typedef enum {
+	BUTTON_IDX_UP     = 0u, //!< The up button.
+	BUTTON_IDX_ENTER  = 1u, //!< The enter button.
+	BUTTON_IDX_DOWN   = 2u, //!< The down button.
+	BUTTON_IDX_RIGHT  = 3u, //!< The right button.
+	BUTTON_IDX_LEFT   = 4u, //!< The left button.
+	BUTTON_IDX_ESCAPE = 5u, //!< The exit (escape) button.
+} BUTTON;
 
 #define NO_OF_BTNS 6  //!< The number of EV3 buttons.
-#define NUM_BUTTONS 6 //!< Number of buttons in the system
+#define NUM_BUTTONS NO_OF_BTNS //!< Number of buttons in the system
+
+//! Button bitmask for aggregating multiple buttons in one value
+typedef enum {
+	BUTTON_ID_UP     = 1u << BUTTON_IDX_UP,     //!< The up button.
+	BUTTON_ID_ENTER  = 1u << BUTTON_IDX_ENTER,  //!< The enter button.
+	BUTTON_ID_DOWN   = 1u << BUTTON_IDX_DOWN,   //!< The down button.
+	BUTTON_ID_RIGHT  = 1u << BUTTON_IDX_RIGHT,  //!< The right button.
+	BUTTON_ID_LEFT   = 1u << BUTTON_IDX_LEFT,   //!< The left button.
+	BUTTON_ID_ESCAPE = 1u << BUTTON_IDX_ESCAPE, //!< The exit (escape) button.
+	BUTTON_ID_ALL = BUTTON_ID_UP |
+			        BUTTON_ID_ENTER |
+                    BUTTON_ID_DOWN  |
+                    BUTTON_ID_RIGHT |
+                    BUTTON_ID_LEFT  |
+                    BUTTON_ID_ESCAPE,           //!< All buttons mask.
+} BUTTON_MASK;
+
+enum {
+	BTN1 = BUTTON_IDX_ESCAPE, //!< The exit (escape) button.
+	BTN2 = BUTTON_IDX_RIGHT,  //!< The right button.
+	BTN3 = BUTTON_IDX_LEFT,   //!< The left button.
+	BTN4 = BUTTON_IDX_ENTER,  //!< The enter button.
+	BTN5 = BUTTON_IDX_UP,     //!< The up button.
+	BTN6 = BUTTON_IDX_DOWN    //!< The down button.
+};
+
+enum {
+	BTNEXIT   = BTN1, //!< The exit (escape) button.
+	BTNRIGHT  = BTN2, //!< The right button.
+	BTNLEFT   = BTN3, //!< The left button.
+	BTNCENTER = BTN4, //!< The enter button.
+	BTNUP     = BTN5, //!< The up button.
+	BTNDOWN   = BTN6, //!< The down button.

```

Comment:
this may break some stuff (https://stackoverflow.com/questions/34677148/why-does-the-c-preprocessor-consider-enum-values-as-equal)

Reply:
solved by adding new enums instead